apps/sms/test/marionette/conversation_test.js

```diff
@@ -491,4 +503,101 @@ marionette('Conversation Panel Tests', function() {
       });
     });
   });
+
+  suite('Add recipients from thread', function() {
+    var conversationView;
+    setup(function() {
+      var thread = ThreadGenerator.generate({
+        numberOfMessages: 1
+      });
+      storage.setMessagesStorage([thread], ThreadGenerator.uniqueMessageId);
+
+      messagesApp.launch();
+
+      var inboxView = new InboxView(client);
+      conversationView = inboxView.goToConversation(thread.id);
+    });
+
+    test('With empty composer: recipients should be kept',
+    function() {
+      var newMessageView = conversationView.addRecipients();
+      // this also ensure we are on Composer view
+      assert.equal(newMessageView.recipients, '+123',
+          'Recipients should be kept');
+      assert.equal(newMessageView.messageText, '', 'SMS text should be empty');
+
+      // add a recipient, a subject and send a mms to create a thread with
+      // several recipients
+      newMessageView.typeMessage('message2');
+      newMessageView.addNewRecipient('+234');
+
+      newMessageView.showSubject();
+      newMessageView.typeSubject('subject1');
+      conversationView = newMessageView.send();
+
+      newMessageView = conversationView.addRecipients();
+      assert.equal(newMessageView.recipients.length, 2,
+        'There should be 2 recipients');
+      assert.include(newMessageView.recipients, '+123',
+        '+123 should be in the recipients');
+      assert.include(newMessageView.recipients, '+234',
+        '+234 should be in the recipients');
```

Comment:
nit: it's quite costly to access .recipients, maybe you can cache it in a variable before asserting ?

Also you can use `assert.lengthOf` instead of `assert.equal(....length, `